client/blocks/free-to-paid-banner/index.jsx

```diff
@@ -0,0 +1,41 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Banner from 'components/banner';
+import {
+	PLAN_PERSONAL,
+} from 'lib/plans/constants';
+import { abtest } from 'lib/abtest';
+import {
+	eligibleForFreeToPaidUpsell,
+} from 'state/selectors';
+
+const FreeToPaidBanner = ( props ) => {
+	if ( ! props.eligibleForFreeToPaidUpsell || abtest( 'freeToPaidUpsell' ) !== 'banner' ) {
+		return null;
+	}
+
+	return (
+		<Banner
+			event="free-to-paid-stats-nudge"
+			plan={ PLAN_PERSONAL }
+			icon="star"
+			title="Choose a free custom domain name with a WordPress.com plan!"
```

Comment:
Not strictly since the abtest will only target en locale at the moment.

I can add a comment to make this clear in case the component is used without an a/b test later.